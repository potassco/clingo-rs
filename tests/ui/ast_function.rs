// ast_function
use clingo::*;

fn main() {
    let loc = ast::Location::default();
    let name = String::from("name");
    let sym = Symbol::create_id("test", true).unwrap();
    let term1: ast::Term = ast::symbolic_term(&loc, &sym).unwrap().into();
    let term2: ast::Term = ast::symbolic_term(&loc, &sym).unwrap().into();
    let args = vec![term1, term2];
    let fun = ast::function(&loc, &name, &args, false).unwrap();
    drop(args);
    println!("{:?}", fun);
}
