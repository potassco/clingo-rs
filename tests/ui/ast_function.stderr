error[E0505]: cannot move out of `args` because it is borrowed
  --> tests/ui/ast_function.rs:12:10
   |
10 |     let args = vec![term1, term2];
   |         ---- binding `args` declared here
11 |     let fun = ast::function(&loc, &name, &args, false).unwrap();
   |                                          ----- borrow of `args` occurs here
12 |     drop(args);
   |          ^^^^ move out of `args` occurs here
13 |     println!("{:?}", fun);
   |                      --- borrow later used here
